Convert README into Markdown

bobvanderlinden opened the pull request:
As per #4, I converted the README to markdown and added information about the new `HOSTS_FILE` environment variable.

Let me know if the conversion to markdown is desired.

figiel commented on README.md:
```diff
@@ -0,0 +1,50 @@
+Some variants of libc have support for the `HOSTALIASES` environment variable.
+This is similar to using `HOSTS_FILE` in this library.
+However, the variant of glibc used on Debian and ArchLinux, amongst others, do not have this feature.
+
```
Let's remove this whole chapter, I don't think we want to keep this up to date with regards to what's implemented elsewhere.

figiel commented on README.md:
```diff
@@ -0,0 +1,50 @@
+
+## Limitations
+
+`hosts` does not work with suid programs due to suid programs clearing the environment before starting. That includes `ping` unfortunately.
```
That's not exactly true (it's a feature of loader to ignore LD_PRELOAD in this case), let's drop "due to suid programs clearing the environment before starting".

figiel commented on README.md:
```diff
@@ -0,0 +1,50 @@
+Allow user-defined hosts entries.
+
+`hosts` is a `LD_PRELOAD` library that overrides the fetching of hostnames in applications
+with functionality that fetches these entries from a user-defined hosts file instead of `/etc/hosts`.
```
I don't think this is clear or accurate, hostnames aren't fetched, they are resolved to IP addresses. I think more precise description wouldn't hurt if you are doing the rewrite. The audience for this is rather deeply technical anyway.

figiel reviewed (commented):
I'm fine with conversion to markdown, with the ~rewrite too, but please take a look at the comments.

bobvanderlinden commented:
Thanks for the comments! I've worked on the readme some more. Incorporated your comments. I also added an example of name resolving redirection (resolve `somehost` using `localhost`, instead of `127.0.0.1`). Let me know what you think :+1:

figiel commented:
Hi, sorry I can't see the changes compared to the version you sent initially. Did you forget to push?

bobvanderlinden commented:
Absolutely right :+1: Pushed them now.
